### PR TITLE
[Java] Refactor FormatUtil class

### PR DESCRIPTION
**Pull request by Bill7zz**
- Import and update the methods from `DateTimeFormatUtil` to `FormatUtil` class keeping them with the proper implementation. 
- Remove duplicated `DateTimeFormatUtil` class
- Refactor `FormatUtil` class name to `DateTimeFormatUtil` to keep the same pattern used in other languages

Build and tests passed correctly
image

### CODE REVIEW

**Review by gasper-az — APPROVED** [review]
LGTM

File: Java/linting-rules.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
             <property name="allowNonPrintableEscapes" value="true"/>
         </module>
         <module name="LineLength">
-            <property name="max" value="180"/>
```
Comment (JuanAr) [inline Java/linting-rules.xml]: Is there a way to format the code with line breaks to pass this validation instead of changing it?

**Review by JuanAr — CHANGES_REQUESTED** [review]
I would prefer to not change the linting rules

File: Java/linting-rules.xml (hunk 1 above)
Comment (Bill7zz) [reply Java/linting-rules.xml]: The line exceeds the limit by one character. The possibles solutions are
- Split the method parameters 
- Remove an indentation tab to keep the same format of the other method calls

Comment (Bill7zz) [reply Java/linting-rules.xml]: image

Comment (JuanAr) [reply Java/linting-rules.xml]: Which file is this one?

Comment (JuanAr) [reply Java/linting-rules.xml]: I think indentation is wrong right now. I used Reformat Code in IntelliJ and it reduced the indentation by one tab (4 spaces). if that fixes the issue, go ahead. If not, add the method argument splitting.
image

Comment (Bill7zz) [reply Java/linting-rules.xml]: BaseDateTimePeriodParser.java 
Line 94 is breaking the rule.

**Review by JuanAr — APPROVED** [review]

**Pull request closed without merging**